site/app/controllers/student/TeamController.php

```diff
@@ -6,9 +6,21 @@
 use app\controllers\admin\AdminGradeableController;
 use app\libraries\FileUtils;
 use app\libraries\response\RedirectResponse;
+use app\models\Team;
 use Symfony\Component\Routing\Annotation\Route;
 
 class TeamController extends AbstractController {
+    /**
+     * Check if a string has special characters.
+     * @param string $str
+     * @return bool
+     */
+    public static function hasSpecialCharacters($str) {
```

Comment:
```suggestion
    /**
     * Check if a string has special characters.
     */
    public static function hasSpecialCharacters(string $str): bool {
```

Reply:
This comment still needs to be addressed.  It is better to use native PHP type annotations whenever possible since they are enforced by the language, rather than PHPDoc comments which are not.